crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/spoof.rs

```diff
@@ -1,8 +1,35 @@
 use crate::state::{start_cheat, stop_cheat, CheatSpan, CheatTarget};
 use crate::CheatnetState;
 use cairo_felt::Felt252;
+use num_traits::ToPrimitive;
+use runtime::utils::buffer_reader::{BufferReadError, BufferReadResult, BufferReader};
+use runtime::utils::from_reader::FromReader;
+use runtime::FromReader;
+use std::ops::Deref;
 
-#[derive(Clone, Default, Debug)]
+#[derive(Clone, Default, Debug, PartialEq)]
+pub struct ResourceBounds(pub Vec<Felt252>);
+
+impl Deref for ResourceBounds {
+    type Target = Vec<Felt252>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl FromReader for ResourceBounds {
+    fn from_reader(reader: &mut BufferReader<'_>) -> BufferReadResult<Self> {
+        let count = reader.read_felt()?;
+
+        let count = count.to_usize().ok_or(BufferReadError::ParseFailed)?;
+        let result = reader.read_slice(count * 3)?; // ResourceBounds struct has 3 fields
+
+        Ok(ResourceBounds(result.to_owned()))
+    }
+}
```

Comment:
Why isn't this derivable? 🤔

Reply:
Because it want to read 3 felts instead of 1
>         let result = reader.read_slice(count * 3)?; // ResourceBounds struct has 3 fields